src/delta.rs

```diff
@@ -458,25 +465,35 @@ fn handle_hunk_header_line(
                 Some(false),
             );
             painter.output_buffer.pop(); // trim newline
-            draw_fn(
-                painter.writer,
-                &painter.output_buffer,
-                &painter.output_buffer,
-                &config.decorations_width,
-                config.hunk_header_style,
-                decoration_ansi_term_style,
-            )?;
-            if !config.hunk_header_style.is_raw {
-                painter.output_buffer.clear()
-            };
+            if !painter.output_buffer.is_empty() {
+                draw_fn(
+                    painter.writer,
+                    &painter.output_buffer,
+                    &painter.output_buffer,
+                    &config.decorations_width,
+                    config.hunk_header_style,
+                    decoration_ansi_term_style,
+                )?;
+            }
+            painter.output_buffer.clear();
         }
     };
+
+    if config.color_only && !config.hunk_header_style.is_raw {
+        painter.emit()?;
+        writeln!(
+            painter.writer,
+            "{}",
+            format::format_raw_line(&raw_line, config)
+        )?;
+    }
+
```

Comment:
Similarly, I believe that if we override options in set.rs this code will not be needed.

Reply:
Yes, we don't need this anymore.

Reply:
Fixed here 8a5fb65